v2/pkg/engine/resolve/resolve.go

```diff
@@ -422,10 +424,14 @@ func (r *Resolver) handleRemoveSubscription(id SubscriptionIdentifier) {
 		trig := r.triggers[u]
 		for ctx, s := range trig.subscriptions {
 			if s.id == id {
-				s.mux.Lock()
-				s.writer.Complete()
-				s.writer = nil
-				s.mux.Unlock()
+
+				if ctx.Context().Err() == nil {
```

Comment:
You could do this check above the loop.
If we're closing 10k subscriptions, it's unnecessary.